deployment/common/types/types.go

```diff
@@ -23,3 +25,58 @@ type MCMSWithTimelockConfig struct {
 	TimelockExecutors []common.Address
 	TimelockMinDelay  *big.Int
 }
+
+type OCRParameters struct {
+	DeltaProgress                           time.Duration
+	DeltaResend                             time.Duration
+	DeltaInitial                            time.Duration
+	DeltaRound                              time.Duration
+	DeltaGrace                              time.Duration
+	DeltaCertifiedCommitRequest             time.Duration
+	DeltaStage                              time.Duration
+	Rmax                                    uint64
+	MaxDurationQuery                        time.Duration
+	MaxDurationObservation                  time.Duration
+	MaxDurationShouldAcceptAttestedReport   time.Duration
+	MaxDurationShouldTransmitAcceptedReport time.Duration
+}
+
+func (params OCRParameters) Validate() error {
```

Comment:
libocr already has exhaustive validation for this, can we call out to that instead of doing this here?

Reply:
I created a ticket to handle this - https://smartcontract-it.atlassian.net/browse/CCIP-4332